cloudinit/sources/DataSourceAzure.py

```diff
@@ -789,9 +790,12 @@ def _negotiate(self):
 
     @azure_ds_telemetry_reporter
     def activate(self, cfg, is_new_instance):
-        address_ephemeral_resize(is_new_instance=is_new_instance,
-                                 preserve_ntfs=self.ds_cfg.get(
-                                     DS_CFG_KEY_PRESERVE_NTFS, False))
+        try:
+            address_ephemeral_resize(is_new_instance=is_new_instance,
+                                     preserve_ntfs=self.ds_cfg.get(
+                                         DS_CFG_KEY_PRESERVE_NTFS, False))
+        finally:
+            push_log_to_kvp()
```

Comment:
I think we need to pass in self.sys_cfg['def_log_file'] which is the merged system config value from any overrides custom images have added to /etc/cloud/cloud.cfg.d/*cfg
```suggestion
            push_log_to_kvp(self.sys_cfg['def_log_file'])
```